src/common/components/game/FullscreenToggle.js

```diff
@@ -1,21 +1,19 @@
 import React, { Component } from 'react';
-import ReactDOM from 'react-dom';
+import { object } from 'prop-types';
 import FontIcon from 'material-ui/FontIcon';
 import IconButton from 'material-ui/IconButton';
 import screenfull from 'screenfull';
 
 import Tooltip from '../Tooltip';
 
 export default class FullscreenToggle extends Component {
+  static propTypes = {
+    gameArea: object
+  }
+
   handleClick = () => {
     this.forceUpdate();
-
-    // TODO come up with a better approach ...
-    /* eslint-disable react/no-find-dom-node */
-    const gameArea = ReactDOM.findDOMNode(this).parentNode.parentNode.parentNode.parentNode.parentNode;
-    /* eslint-enable react/no-find-dom-node */
-
-    screenfull.toggle(gameArea);
+    screenfull.toggle(this.props.gameArea);
```

Comment:
It doesn't make sense to me that this isn't calling a method on the gameArea. It seems like it would be the responsibility of gameArea to know how to toggle itself and then the method that does that is passed in via the props instead of the full gameArea.

```javascript
this.props.toggleFullscreen();
```

That would also make this component more composable with other components that might want to make themselves fullscreen.